src/System.Security.Permissions/src/System/Security/SecurityElement.cs

```diff
@@ -2,30 +2,827 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.s
 
+using System.Collections;
+using System.Text;
+using System.Globalization;
+using System.IO;
+using System.Diagnostics;
+
 namespace System.Security
 {
+    internal enum SecurityElementType
+    {
+        Regular = 0,
+        Format = 1,
+        Comment = 2
+    }
+
+    internal interface ISecurityElementFactory
+    {
+        SecurityElement CreateSecurityElement();
+
+        object Copy();
+
+        string GetTag();
+
+        string Attribute(string attributeName);
+    }
+
     [Serializable]
-    public sealed partial class SecurityElement
+    sealed public class SecurityElement : ISecurityElementFactory
     {
-        public SecurityElement(string tag) { }
-        public SecurityElement(string tag, string text) { }
-        public System.Collections.Hashtable Attributes { get; set; }
-        public System.Collections.ArrayList Children { get; set; }
-        public string Tag { get; set; }
-        public string Text { get; set; }
-        public void AddAttribute(string name, string value) { }
-        public void AddChild(SecurityElement child) { }
-        public string Attribute(string name) { return default(string); }
-        public SecurityElement Copy() { return this; }
-        public bool Equal(SecurityElement other) { return false; }
-        public static string Escape(string str) { return default(string); }
-        public static SecurityElement FromString(string xml) { return default(SecurityElement); }
-        public static bool IsValidAttributeName(string name) { return false; }
-        public static bool IsValidAttributeValue(string value) { return false; }
-        public static bool IsValidTag(string tag) { return false; }
-        public static bool IsValidText(string text) { return false; }
-        public SecurityElement SearchForChildByTag(string tag) { return default(SecurityElement); }
-        public string SearchForTextOfTag(string tag) { return default(string); }
-        public override string ToString() => base.ToString();
+        internal string m_strTag;
+        internal string m_strText;
+        private ArrayList m_lChildren;
+        internal ArrayList m_lAttributes;
+
+        internal readonly SecurityElementType m_type = SecurityElementType.Regular;
+        private const int c_AttributesTypical = 4 * 2;  // 4 attributes, times 2 strings per attribute
+        private const int c_ChildrenTypical = 1;
```

Comment:
Nit: constants don't need `c_` prefix should just be `AttributesTypical` and `ChildrenTypical`.